iup/__init__.py

```diff
@@ -831,3 +846,446 @@ def predict(
         self.cumulative_projection = CumulativeUptakeData(self.cumulative_projection)
 
         return self
+
+
+#### prediction output ####
+class Forecast(ValidateData):
+    """
+    Abstract class for all the forecast data type.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.validate()
+
+    # Can add validate to enforce the specific schema of Forecast when more Forecast types are added
+    def validate(self):
+        self.assert_type_included(pl.Date)
+        self.assert_type_included(pl.Float64)
+        # has at least 1 column of date and 1 column of estimate
+
+    def assert_type_included(self, datatype):
```

Comment:
Why not use the `assert_columns_type` function I have already written? It seems to do the same thing.

Reply:
If we have 1000 columns for 1000 draws from the posterior, then these columns ought to have well-specified names (e.g. "estimate_1", "estimate_2", etc.). So we could just construct a list of these column names to give to `assert_columns_type`. But again, maybe this is something I will address in the future.